weblab-identity-auth0/src/test/java/ucles/weblab/common/identity/auth0/UserAttributesResolverTest.java

```diff
@@ -46,7 +46,26 @@ public void shouldReturnOAuth2WhenGetId() {
         UserAttributesResolver userAttributesResolver = new UserAttributesResolver(oAuth2Authentication);
 
         when(oAuth2Authentication.getUserAuthentication()).thenReturn(authentication);
-        when(authentication.getName()).thenReturn(someUserId);
+        when(authentication.getPrincipal()).thenReturn(principal);
+        when(principal.toString()).thenReturn(someUserId);
+
+        assertEquals(userAttributesResolver.getUserId(), someUserId);
+    }
+
+    @Test
+    public void shouldReturnOAuth2ExtendedUserWhenGetId() {
```

Comment:
I think you should include a test which demonstrates the undesirable behaviour of the previous version of the code. You've got one that would fail because it doesn't pick up the 'sub' attribute, but it would be nice to also mock authentication.getName() with the username, so we have documented the behaviour we DON'T want, to avoid someone deciding that's a nice refactoring in future.

Reply:
I agree to this. I pushed changes where some test case must call certian methods and for other's what to avoid calling.